### DISCUSSION TITLE
API Filter to pull a CIDR block of IPs or match in first few octets

### DISCUSSION THREAD

**Discussion by bradbendy** [netbox-community/netbox#17080] [Help Wanted!]
Hi,

I've looked everywhere and can't find anything on this. I want to pull all the IP addresses in say a single /24. I don't know of a filter to do this one. 

I do see this in another post: https://github.com/netbox-community/netbox/blob/v3.1.6/netbox/ipam/filtersets.py#L589-L593

But when I add address__istartwith=x.y.z i get no difference in results. I also tried address__net_in=x.y.z.0/24 for example and same thing.

I was testing with netbox.domain.com/api/ipam/ip-addresses/?address__net_in= for exmaple

Am I missing something?

Thanks

**Comment by candlerb** [comment]
I think what you are looking for is the `parent` filter on IP address:

```
/api/ipam/ip-addresses/?parent=x.y.z.0/24
```

Source references [here](https://github.com/netbox-community/netbox/blob/v4.0.8/netbox/ipam/filtersets.py#L561-L564) and [here](https://github.com/netbox-community/netbox/blob/v4.0.8/netbox/ipam/filtersets.py#L690-L700).

In the Django ORM, you'd use:

```
IPAddress.objects.filter(address__net_host_contained="x.y.z.0/24")
```

(example in [this custom script](https://github.com/netbox-community/customizations/pull/94/files))

Note that neither of these depend on an actual Prefix object existing in the database. It's just filtering on IP addresses which sit within a particular CIDR block.

**Reply by bradbendy** [reply]
Ahh there we go! Next time I will dig into the filterset area. I knew it was somehwhere!

Thanks much
